onnxruntime/core/framework/data_transfer.cc

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/data_transfer.h"
+
+namespace onnxruntime {
+
+common::Status IDataTransfer::CopyTensor(const Tensor& src, Tensor& dst) const {
+  return CopyTensor(src, dst, 0);
+}
+
+bool CPUDataTransfer::CanCopy(const OrtDevice& src_device, const OrtDevice& dst_device) const {
+  return src_device.Type() == OrtDevice::CPU && dst_device.Type() == OrtDevice::CPU;
+}
+
+common::Status CPUDataTransfer::CopyTensor(const Tensor& src, Tensor& dst, int /*exec_queue_id*/) const {
+  size_t bytes = src.DataType()->Size() * src.Shape().Size();
+  const void* src_data = src.DataRaw();
```

Comment:
I believe src.Size() does this. 

This frequently comes up and makes me wonder if Tensor::Size should be rename to something like Tensor::SizeInBytes or something like that.

Reply:
Yes, I've heard this multiple times. Size() in C++ conventionally returns the number of elements, not the bytes. This is really bad naming. We should rename this to NumBytes to be super clear.